### PR TITLE
[server][wmts][fix] updated constant to correct resolution in WMTS tile grid initialization

### LINKED ISSUE #34826: QGIS Server: WMTS only in CRS 3857
Hi!

It seams that QGIS server can produce WMTS in CRS 3857 only. I am using QGIS server 3.10 with FastCGI and QGIS Desktop 3.10 on Windows server 2012.

I wonder if this is a known issue?

Best Regards.
Abel

### LINKED ISSUE #39040: QGIS-server WMTS custom tilematrix set: mismatch between original data and wmts version
[28992.zip](https://github.com/qgis/QGIS/files/5283458/28992.zip)

In that zip is data (a simple geopackage) AND a project file. Loading it in QGIS you should see:

Screenshot-20200925164508-1015x520

From the official document:
https://www.geonovum.nl/uploads/standards/downloads/nederlandse_richtlijn_tiling_-_versie_1.1.pdf
AND from our national service WMTS url:
https://geodata.nationaalgeoregister.nl/wmts?VERSION=1.0.0&request=GetCapabilities
I got the following properties for the WMTS (and edited those in the Grids table):
EPSG:28992
Top: 903402.00  (these are meters)
Left: -285401.92
Min.Scale: 12288000.00
Last level 14

Creating a QGIS server from this results in the right tilematrix:

```
<TileMatrixSet><ows:Identifier>EPSG:28992</ows:Identifier><ows:SupportedCRS>EPSG:28992</ows:SupportedCRS><TileMatrix><ows:Identifier>0</ows:Identifier><ScaleDenominator>12288000</ScaleDenominator><TopLeftCorner>-285401.92 903402</TopLeftCorner><TileWidth>256</TileWidth><TileHeight>256</TileHeight><MatrixWidth>1</MatrixWidth><MatrixHeight>1</MatrixHeight></TileMatrix>
```

as from the offical pdok service:

```
<TileMatrixSet><ows:Identifier>EPSG:28992</ows:Identifier><ows:SupportedCRS>urn:ogc:def:crs:EPSG::28992</ows:SupportedCRS><TileMatrix><ows:Identifier>00</ows:Identifier><ScaleDenominator>1.2288E7</ScaleDenominator><TopLeftCorner>-285401.92 903402.0</TopLeftCorner><TileWidth>256</TileWidth><TileHeight>256</TileHeight><MatrixWidth>1</MatrixWidth><MatrixHeight>1</MatrixHeight></TileMatrix>
```

So I think that is ok. Creating the WMTS now and requesting it in the WMS/WMTS dialog show the proper layer and tilesets:

Screenshot-20200925165334-479x362

But then loading the layer shows it is shifted:

Screenshot-20200925165438-861x434

To double check you can add the 'opentopo' layer from this public WMTS:

https://geodata.nationaalgeoregister.nl/wmts?VERSION=1.0.0&request=GetCapabilities

As you can see the wmts seems shifted:

Screenshot-20200925165624-1103x488

NOTE: we did have this kind of shifts earlier, when we used different proj params for EPSG:28992, and the default epsg params where not right yet in the official db...

But as you can see the data itself is exactly mapping the PDOK WMTS. and I would think the project/data and the server/data would use the same epsg:28992 definitions?

### LINKED ISSUE #42942: QGIS Server: Published WMTS layer displayed with offset
Hi,

i'm using QGIS Server 3.16.6 and am trying to publish an orthophoto layer as WMTS. Unfortunately, when i feed the WMTS layer  back into the original project used for publishing it there is a significant offset between the original orthophoto layer and the tiles from service.

I have been trying for some time to figure out if i have configured something wrong .... but came up with nothing. Hence my attempt to post this as an issue ... maybe even a bug.

I've attached ...

- the QGIS project containing the DOP layer i want to publish
- WMTS GetCapabilities document produced by QGIS Server
- screenshot displaying the offset between the original DOP and the tiles published through QGIS Server
- GeoPackage with the coordinates of the reference points displayed in the screenshot

[2021-04-25_issue_project_data.zip](https://github.com/qgis/QGIS/files/6371675/2021-04-25_issue_project_data.zip)

### PR DESCRIPTION
**Pull request by andreassteffens**
Constant needed to be updated in order to match the standard rendering pixel size of 0.28mm x 0.28mm defined by the OGC resulting in DPI of 90.71428571429.

Should fix #34826, fix #42942, fix #39040

### CODE REVIEW

File: src/server/services/wmts/qgswmtsutils.cpp [hunk 1]
```diff
@@ -36,7 +36,7 @@ namespace QgsWmts
 
     // Constant
     int tileSize = 256;
-    double POINTS_TO_M = 2.83464567 / 10000.0;
+    double POINTS_TO_M = 2.800005600011068 / 10000.0;
```
Comment (elpaso) [inline src/server/services/wmts/qgswmtsutils.cpp:39]:
```suggestion
    const double POINTS_TO_M = 2.800005600011068 / 10000.0;
```

**Review by rldhont — CHANGES_REQUESTED** [review]
thanks @andreassteffens can you also const tileSize ?

**Comment by DelazJ** [comment]
> Should fix #34826, #42942, #39040

@andreassteffens if you want the three to be closed, the "fix" mention should be added for all, ie `Should fix #34826, fix #42942, fix #39040`

**Review by rldhont — APPROVED** [review]

**Comment by rldhont** [comment]
@andreassteffens can you run the tests and update the WMTS GetCapabilities that failed.

**Comment by andreassteffens** [comment]
will take a look at the test case ASAP

**Pull request merged**